iml-agent/src/daemon_plugins/snapshot.rs

```diff
@@ -87,14 +88,37 @@ async fn list() -> Result<Vec<Snapshot>, ()> {
     let really_failed_fss = errs
         .into_iter()
         .map(|x| x.unwrap_err())
-        .filter(|x| !snapshot_fsnames.contains(&x.0))
+        .filter(|x| {
+            tracing::debug!("listing for {} failed: {:?}", x.0, x.1);
+
+            if snapshot_fsnames.contains(&x.0) {
+                tracing::debug!("{} is a snapshot FS", x.0);
+                return false;
+            }
+
+            match &x.1 {
+                ImlAgentError::CmdError(CmdError::Output(o)) => {
+                    // XXX lctl returns 1 no matter what, so have to read its output:
+                    let stderr = String::from_utf8_lossy(&o.stderr);
+                    if stderr.find("Resource temporarily unavailable").is_some() {
+                        true
+                    } else if stderr.find("Miss MDT0 in the config file").is_some() {
```

Comment:
Clippy warning here

Reply:
Maybe the first time I am inclined to disagree with clippy, but there is a better way though.